trafaret/base.py

```diff
@@ -486,7 +487,7 @@ def __repr__(self):
 
 
 class Bytes(String):
-    str_type = (BYTES_TYPE,)
+    str_type = BYTES_TYPE
```

Comment:
I'm not sure that `if` is good way. If u in future will have another subclass of `String` so u need to add new if else.

Instead of that i recommend use some like this:

```python
class String:
    TYPE_ERROR_MESSAGE = "value is not a string"
    TYPE_ERROR_CODE = codes.IS_NOT_A_STRING

    def check_and_return(self, value):
        ...
        self._failure(self.TYPE_ERROR_MESSAGE, value=value, code=self.TYPE_ERROR_CODE)
        
```

and when u will create a new subclass u need just override this properties.

```python
class String(String):
    str_type = (BYTES_TYPE,)

    TYPE_ERROR_MESSAGE = "value is not a bytes string"
    TYPE_ERROR_CODE = codes.IS_NOT_A_BYTES_STRING

```

Also, u forgot that code of the error has the meaning too.

Reply:
Yeah, I agree with you. Thought that `Bytes` and `AnyString` are the only subclasses of `String`  in this project. I suggest then changing the error message for the class `String`:
```python
TYPE_ERROR_MESSAGE = "value is not a pure string"
TYPE_ERROR_CODE = codes.IS_NOT_A_PURE_STRING
```
and for subclass `AnyString`:
```python
TYPE_ERROR_MESSAGE = "value is not a string"
TYPE_ERROR_CODE = codes.IS_NOT_A_STRING
```
Because when we do so, the message is unclear.
```python
In [1]: t.String().check(b'test')
... 
DataError: value is not a string
```

Reply:
I think it's not necessary.